continuous_integration/scripts/setup-utils

```diff
@@ -0,0 +1,268 @@
+        if [[ -z ${ver2[i]} ]]; then
+            # ver1 has more segments and is greater
+            set -x
+            return 0
+        fi
+
+        if ((10#${ver1[i]} > 10#${ver2[i]})); then
+            set -x
+            return 0
+        elif ((10#${ver1[i]} < 10#${ver2[i]})); then
+            set -x
+            return 1
+        fi
+    done
+
+    return 0
+}
+
+install_from_apt() {
+    set -xeuo pipefail
+
+    export DEBIAN_FRONTEND=non-interactive
+
+    # Run package updates and install packages
+    apt-get update
+    apt-get install -y wget curl jq sudo ninja-build vim numactl rsync
+}
+
+install_sccache_linux() {
+    set -xeuo pipefail
+
+    wget https://github.com/mozilla/sccache/releases/download/v0.5.4/sccache-v0.5.4-x86_64-unknown-linux-musl.tar.gz && \
+    tar -xf sccache-v0.5.4-x86_64-unknown-linux-musl.tar.gz && \
+    sudo mv sccache-v0.5.4-x86_64-unknown-linux-musl/sccache /usr/bin/sccache
+}
+
+maybe_install_sccache_linux() {
+    set -xeuo pipefail
+
+    if ! command -v sccache &> /dev/null; then
+        echo "sccache not found, proceeding with installation."
+        install_sccache_linux
+    else
+        sccache_version=$(sccache --version 2>&1 | awk '/sccache/ {print $2}')
+        if [[ -z "$sccache_version" ]] || ! version_greater_equal "$sccache_version" "0.5.4"; then
+            echo "sccache version less than 0.5.4, proceeding with installation."
+            install_sccache_linux
+        else
+            echo "sccache version is 0.5.4 or greater, no need to install."
+        fi
+    fi
+}
+
+
+install_cmake() {
+    set -xeuo pipefail
+
+    wget https://github.com/Kitware/CMake/releases/download/v3.26.5/cmake-3.26.5-linux-x86_64.tar.gz
+
+    tar -xzf cmake-3.26.5-linux-x86_64.tar.gz
+}
+
+setup_linux_build_env() {
+    set -xeuo pipefail
+    export OS_SHORT_NAME=linux
+    export PATH="${PATH}:${PREBUILD_DIR}/cmake-3.26.5-linux-x86_64/bin"
+
+
+    mkdir -p /tmp/out /tmp/env_yaml
+}
+
+install_linux_tools() {
+    set -xeuo pipefail
+
+    export SED=sed
+    export READLINK=readlink
+
+    install_from_apt;
+    maybe_install_sccache_linux;
+    install_cmake;
+
+    mkdir -p /tmp/out /tmp/env_yaml
+}
+
+install_linux_test_tools() {
+    set -xeuo pipefail
+
+    export SED=sed
+    export READLINK=readlink
+}
+
+prep_git() {
+    local current_email=$(git config --global user.email)
+    local current_name=$(git config --global user.name)
+
+    if [ -z "$current_email" ]; then
+        git config --global --add user.email "users.noreply.github.com"
+    else
+        echo "Note: git user.email is already set to $current_email"
+    fi
+
+    if [ -z "$current_name" ]; then
+        git config --global --add user.name "anon"
+    else
+        echo "Note: git user.name is already set to $current_name"
+    fi
+
+    # Fix "fatal: detected dubious ownership in repository at '/tmp/legate.core'"
+    # during local builds.
+    git config --global --add safe.directory "$REPO_DIR"
+}
+
```

Comment:
Can you add a seperator line here, since everything above is OS-specific helper functions, everything below is a generalized function?

Reply:
Done.